modules/repository-datastax/src/main/java/com/intuit/wasabi/repository/cassandra/impl/CassandraExperimentRepository.java

```diff
@@ -1253,8 +1249,13 @@ public void createApplication(Application.Name applicationName) {
 
             Map<Application.Name, Set<String>> result = new HashMap<>();
             for (ListenableFuture<Result<ExperimentTagsByApplication>> future : futures) {
-                ExperimentTagsByApplication expTagApplication = future.get().one();
-                result.put(Application.Name.valueOf(expTagApplication.getAppName()), expTagApplication.getTags());
+                List<ExperimentTagsByApplication> expTagApplication = future.get().all();
+
+                Set<String> allTagsForApplication = new TreeSet<>();
```

Comment:
@AndreaSuckro Why use a `TreeSet`? Can we not use a `HashSet()` instead? Do we need any sorted order here? `TreeSet` is expensive in terms of performance (both insert and retrieve)